crates/db_schema/src/impls/local_site.rs

```diff
@@ -14,9 +18,21 @@ impl LocalSite {
       .get_result::<Self>(conn)
       .await
   }
-  pub async fn read(pool: &mut DbPool<'_>) -> Result<Self, Error> {
-    let conn = &mut get_conn(pool).await?;
-    local_site.first::<Self>(conn).await
+  pub async fn read(pool: &mut DbPool<'_>) -> Result<Self, LemmyError> {
+    static CACHE: Lazy<Cache<(), LocalSite>> = Lazy::new(|| {
+      Cache::builder()
+        .max_capacity(1)
+        .time_to_live(Duration::from_secs(1))
+        .build()
+    });
```

Comment:
Should the static be defined in the function? I thought stuff like this was usually handled at the module level.

Reply:
Its only used in this one function, so no need to make it available anywhere else.

Reply:
why not extend the time to live, and invalidate cache in the 'update' method? 1 sec is really short

Reply:
Because some Lemmy instances may run more than one backend for horizontal scaling. So the value can be updated through another backend and we wont know about it unless we read again from the db.

Reply:
I grepped the other `time_to_live` , and like the others it'd be good to extract this into a const somewhere.

Reply:
Then we should really inject a cache service into the lemmy instance, so people can swap between something like redis if they have scaling, or just in-mem cache, and not have to worry about invalidation. Probably outside the scope of this pr though.

Reply:
that would be pointless - the query is already a simple single select postgresql query and redis is not faster for a single simple select than pg

Reply:
theres more than just the localsite being cached already, and theres lots that can be added. And yeah, redis would be slower, but inmemory cache would be faster. Both reduce db pressure, though, which is sorely needed.

Reply:
improving expensive db queries is needed. removing 0.1ms db queries will not make a dent unless called thousands of times per second

Reply:
Does the time to live make it so the cache is invalidated every second?

Reply:
yes, as is this will make the number of queries for local site go to around 1 per second for an active instance

Reply:
I'm surprised that 1 second is enough to improve performance. This must be called a lot of times per second.

Reply:
Lemmy.ml receives roughly 230 votes per minute, so it will be called at least that many times. Actually more for post creations and other actions. Still it probably wont make a noticable difference unless you aggressively [cache api responses](https://github.com/LemmyNet/lemmy-ansible/issues/195).